src/dra/7A4D0.c

```diff
@@ -93,36 +90,36 @@ void func_8011A4D0(void) {
             continue;
         }
         if (entity->step == 0) {
-            entityId = entity->entityId;
-            if (entityId < 0xD0) {
+            if (entity->entityId < 0xD0) {
                 // Objects 00-CF
-                entity->pfnUpdate = g_DraEntityTbl[entityId];
-            } else if (entityId < 0xE0) {
+                entity->pfnUpdate = g_DraEntityTbl[entity->entityId];
+            } else if (entity->entityId < 0xE0) {
                 // Objects D0-DF
-                entity->pfnUpdate = D_8016FCC0[entityId];
-            } else if (entityId == 0xEF || entityId == 0xFF ||
-                       entityId == 0xED || entityId == 0xFD) {
+                entity->pfnUpdate =
+                    ((PfnEntityUpdate*)FAMILIAR_PTR)[entity->entityId - 0xD0];
+            } else if (entity->entityId == 0xEF || entity->entityId == 0xFF ||
+                       entity->entityId == 0xED || entity->entityId == 0xFD) {
                 entity->pfnUpdate = g_DraEntityTbl[1];
-            } else if (entityId == 0xEE || entityId == 0xFE) {
+            } else if (entity->entityId == 0xEE || entity->entityId == 0xFE) {
                 entity->pfnUpdate = g_DraEntityTbl[15];
-            } else if (entityId >= 0xF0) {
+            } else if (entity->entityId >= 0xF0) {
                 // Objects F0-FC
                 entity->pfnUpdate =
-                    ((PfnEntityUpdate*)(&D_8017D000))[entityId - 0xF0];
+                    ((PfnEntityUpdate*)WEAPON1_PTR)[entity->entityId - 0xF0];
             } else {
                 // Objects E0-EC
                 entity->pfnUpdate =
-                    ((PfnEntityUpdate*)(&D_8017A000))[entityId - 0xE0];
+                    ((PfnEntityUpdate*)WEAPON0_PTR)[entity->entityId - 0xE0];
             }
         }
         if ((temp_s2 == 0) || (entity->flags & FLAG_UNK_10000)) {
             entity->pfnUpdate(entity);
             entity = g_CurrentEntity;
             if (entity->entityId != 0) {
                 if (!(entity->flags & FLAG_UNK_04000000) &&
-                    ((u16)(entity->posX.i.hi + 32) > 320 ||
-                     (u16)(entity->posY.i.hi + 16) > 272)) {
-                    DestroyEntity(entity);
+                    (entity->posX.i.hi > 288 || entity->posX.i.hi < -32 ||
+                     entity->posY.i.hi > 256 || entity->posY.i.hi < -16)) {
+                    DestroyEntity(g_CurrentEntity);
                 } else {
                     if (entity->flags & 0x100000) {
                         UpdateAnim(NULL, D_800ACFB4);
```

Comment:
Not sure how it affects other things but I found that this variable needed to change to 
```
extern  AnimationFrame**  D_800ACFB4;
``` 
to compile with MWCC

Reply:
Yes, that's right, the problem is that `func_80118C28` references individual bytes from that variable and I don't know how to rewrite that function to be acceptable.

Reply:
My scratch is here: https://decomp.me/scratch/UbCHl I couldn't get it to work with `AnimationFrame**`, I had to use `AnimationFrame* D_800ACFB4[]`